src/controllers/aer_controller.hpp

```diff
@@ -819,27 +841,67 @@ Result Controller::execute(const inputdata_t &input_qobj) {
 
 Result Controller::execute(std::vector<Circuit> &circuits,
                            Noise::NoiseModel &noise_model,
-                           const json_t &config) {
+                           const json_t &config) 
+{
   // Start QOBJ timer
   auto timer_start = myclock_t::now();
 
   // Determine simulation method for each circuit
   // and enable required noise sampling methods
   auto methods = simulation_methods(circuits, noise_model);
+  bool one_method = true;
 
   // Initialize Result object for the given number of experiments
   Result result(circuits.size());
 
   // Execute each circuit in a try block
   try {
+    std::vector<bool> multi_chunk(circuits.size());
+    bool multi_chunk_req = false;
     // get max qubits for this process (to allocate qubit register at once)
+    int i_max_circ = 0;
     max_qubits_ = 0;
-    for (size_t j = 0; j < circuits.size(); j++) {
+
+    for (size_t j = 0; j < circuits.size(); j++){
       if (circuits[j].num_qubits > max_qubits_) {
         max_qubits_ = circuits[j].num_qubits;
+        i_max_circ = j;
+      }
+      if(methods[j] != methods[0])
+        one_method = false;
+      if(circuits[j].num_qubits > 0){
+        multi_chunk[j] = multiple_chunk_required(circuits[j], noise_model, methods[j]);
+        if(multi_chunk[j])
+          multi_chunk_req = true;
+      }
+      else{
+        multi_chunk[j] = false;
+      }
+    }
+    if(multi_chunk_req)
+      num_process_per_experiment_ = num_processes_;
+    else
+      num_process_per_experiment_ = 1;
+
+    if(max_qubits_ == 0){
+      max_qubits_ = 1;
+      max_batched_states_ = 1;
+    }
+    else{
+      //set max batched states
+      uint_t max_required = required_memory_mb(circuits[i_max_circ], noise_model, methods[i_max_circ]);
+      if(max_required == 0){
+        max_batched_states_ = 1;
+      }
+      else{
+        if(sim_device_ == Device::GPU){
+          max_batched_states_ = ((max_gpu_memory_mb_/num_gpus_*8/10) / max_required)*num_gpus_;
+        }
+        else{
+          max_batched_states_ = (max_memory_mb_*8/10) / max_required;
+        }
```

Comment:
All the above check codes should be in `set_parallelization_experiments()`, which determines experiments-level configuration.

Reply:
moved to `set_parallelization_experiments()`